packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -546,7 +546,8 @@ class AndroidDevice extends Device {
       observatoryDiscovery = ProtocolDiscovery.observatory(
         await getLogReader(),
         portForwarder: portForwarder,
-        hostPort: debuggingOptions.hostVmServicePort,
+        // DDS will be listening on the hostVmServicePort.
```

Comment:
Can you expand on this comment a bit more?

Reply:
Sure. I'll add this in-line as well, but basically DDS is trying to act as if it is the real VM service without clients needing to know that there's an intermediary there. To make that work, if the user provides a port they want "Observatory" to listen on we actually need to have DDS listen on that port and forward everything to the VM service. Since the user isn't going to see the real VM service URI, we set `hostPort: 0` to bind to a random address since it really doesn't matter which port the VM service is listening on as long as DDS knows which port to use.